src/hotspot/share/opto/predicates.cpp

```diff
@@ -1101,6 +1104,16 @@ void UpdateStrideForAssertionPredicates::visit(const TemplateAssertionPredicate&
   connect_initialized_assertion_predicate(template_tail_control_out, initialized_assertion_predicate);
 }
 
+// Kill the old Initialized Assertion Predicates with old strides before unrolling. The new Initialized Assertion
+// Predicates are inserted after the Template Assertion Predicate which ensures that we are not accidentally visiting
+// and killing a newly created Initialized Assertion Predicate here.
+void UpdateStrideForAssertionPredicates::visit(const InitializedAssertionPredicate& initialized_assertion_predicate) {
+  if (initialized_assertion_predicate.is_last_value()) {
+    // Only Last Value Initialized Assertion Predicates need to be killed and updated.
+    initialized_assertion_predicate.kill(_phase->igvn());
```

Comment:
Needed to move this method to the source file because of incomplete `PhaseIdealLoop` type when trying to access `igvn()`.